vms/platformvm/txs/executor/standard_tx_executor.go

```diff
@@ -1012,6 +1012,114 @@ func (e *StandardTxExecutor) IncreaseBalanceTx(tx *txs.IncreaseBalanceTx) error
 	return nil
 }
 
+func (e *StandardTxExecutor) DisableSubnetValidatorTx(tx *txs.DisableSubnetValidatorTx) error {
+	var (
+		currentTimestamp = e.State.GetTimestamp()
+		upgrades         = e.Backend.Config.UpgradeConfig
+	)
+	if !upgrades.IsEtnaActivated(currentTimestamp) {
+		return errEtnaUpgradeNotActive
+	}
+
+	if err := e.Tx.SyntacticVerify(e.Ctx); err != nil {
+		return err
+	}
+
+	if err := avax.VerifyMemoFieldLength(tx.Memo, true /*=isDurangoActive*/); err != nil {
+		return err
+	}
+
+	sov, err := e.State.GetSubnetOnlyValidator(tx.ValidationID)
+	if err != nil {
+		return fmt.Errorf("couldn't load SoV for %s: %w", tx.ValidationID, err)
+	}
+
+	var disableOwner message.PChainOwner
+	if _, err := txs.Codec.Unmarshal(sov.DeactivationOwner, &disableOwner); err != nil {
+		return err
+	}
+
+	baseTxCreds, err := verifyAuthorization(
+		e.Fx,
+		e.Tx,
+		&secp256k1fx.OutputOwners{
+			Threshold: disableOwner.Threshold,
+			Addrs:     disableOwner.Addresses,
+		},
+		tx.DisableAuth,
+	)
+	if err != nil {
+		return err
+	}
+
+	// Verify the flowcheck
+	fee, err := e.FeeCalculator.CalculateFee(tx)
+	if err != nil {
+		return err
+	}
+
+	if err := e.Backend.FlowChecker.VerifySpend(
+		tx,
+		e.State,
+		tx.Ins,
+		tx.Outs,
+		baseTxCreds,
+		map[ids.ID]uint64{
+			e.Ctx.AVAXAssetID: fee,
+		},
+	); err != nil {
+		return err
+	}
+
+	txID := e.Tx.ID()
+
+	// Consume the UTXOS
+	avax.Consume(e.State, tx.Ins)
+	// Produce the UTXOS
+	avax.Produce(e.State, txID, tx.Outs)
+
+	// If the validator is already disabled, there is nothing to do.
+	if sov.EndAccumulatedFee == 0 {
+		return nil
+	}
+
+	var remainingBalanceOwner message.PChainOwner
+	if _, err := txs.Codec.Unmarshal(sov.RemainingBalanceOwner, &remainingBalanceOwner); err != nil {
+		return err
+	}
+
+	accruedFees := e.State.GetAccruedFees()
+	if sov.EndAccumulatedFee <= accruedFees {
+		// This check should be unreachable. However, including it ensures
```

Comment:
well you should have this check also because of the next line (1098)

Reply:
This check should be impossible to be reached.

We already know that `sov.EndAccumulatedFee != 0` (as otherwise the function would have already returned).

If `sov.EndAccumulatedFee < accruedFees`, then the balance on the SoV is negative, which should clearly never happen.

If `sov.EndAccumulatedFee == accruedFees` then there must have been a bug in `AdvanceTimeTo`, as we should have set the `EndAccumulatedFee` to be equal to `0` there already.

The only reason this check is here is as a last-chance verification to ensure that a bug in execution doesn't cause the AVAX balance to underflow (which could be catastrophic)

Reply:
> The only reason this check is here is as a last-chance verification to ensure that a bug in execution doesn't cause the AVAX balance to underflow (which could be catastrophic)

Yes that's what I meant.